microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import Gpu, GpuEnabled, SerialConsole
+from lisa.tools import Tar, Wget
+
+OSU_MPI_LOCATION = (
+    "https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz"
+)
+OSU_MPI_FILE_NAME = "mvapich2.tgz"
+
+OSU_BENCH_LOCATION = (
+    "http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz"
+)
+OSU_BENCH_FILE_NAME = "osu-bench.tgz"
+
+@TestSuiteMetadata(
+    area="hpc",
+    category="functional",
+    name="OSU Bench",
+    description="""
+    This test suite runs the OSU Micro-Benchmarks MPI test cases.
+    """,
+)
+class OSUTestSuite(TestSuite):
+    TIMEOUT = 200000
+
+    @TestCaseMetadata(
+        description="""
+            This test case runs GPU/CPU MPI latency.
+
+            Steps:
+            1. Install MVAPICH;
+            2. Install OSU Micro_Benchmarks;
+            3. Run GPU/CPU collective/latency tests in a single node.
+
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log: Logger) -> None:
+
+        _install_osu_mpi(node)
+        path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        node.execute(
+            f"mpirun -np {gpu_count} python3 {path}/python/run.py --benchmark allgather"
+            " --min 128 --max 10000000 >> /tmp/osu_bench.log",  # --buffer pycuda #--buffer cupy
+            shell=True,
+            sudo=True,
+        )
+        log.info("OSU MPI tests finish!")
+
+def _install_osu_mpi(node: Node) -> None:
+
+    wget = node.tools[Wget]
+    tar = node.tools[Tar]
+
+    path = wget.get_tool_path(use_global=True)
+    extracted_path = tar.get_tool_path(use_global=True)
+    if node.shell.exists(path / OSU_MPI_FILE_NAME):
+        return
+
+    download_path = wget.get(
+        url=OSU_MPI_LOCATION, filename=str(OSU_MPI_FILE_NAME), file_path=str(path)
+    )
+    tar.extract(download_path, dest_dir=str(extracted_path))
+
+    target_path = "/usr/local/"
+    node.execute(
```

Comment:
You don't need to use `cd`. You can use `cwd` option in the node.execute. This will run the command from the given folder. For example:
```
node.execute("./configure --prefix {target_path}", shell=True, cwd = f"{extracted_path}/mvapich2-2.3.7-1")
```

Further, sudo is false by default. So we don't need to specify default arguments in the function call